csrc/preseg_passes/remove_bcast_squeeze.cpp

```diff
@@ -0,0 +1,72 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <debug.h>
+#include <ir/utils.h>
+#include <options.h>
+#include <preseg_passes/remove_bcast_squeeze.h>
+#include <root_domain_map.h>
+
+namespace nvfuser::preseg_passes {
+
+namespace {
+// Remove broadcast-squeeze and squeeze-broadcast patterns
+// TODO: still remove when have intermediate ops between broadcast and squeeze
+void removeBcastSqueeze(Fusion* fusion) {
+  for (auto expr : fusion->exprs()) {
+    // step-1: find and remove broadcast + squeeze pattern
+    // before: Y = broadcast(X); Z = squeeze(Y);  M = someOp(Z)
+    // after : M = someOp(X)
+    // conditions: (1) broadcast and squeeze have the same dim flags
+    //             (2) Y is only consumed by Z (has only one consumer)
+    // special case: if Z is a fusion output, replace the output with X
+    if (auto squeeze = dynamic_cast<SqueezeOp*>(expr)) {
+      if (auto bcast =
+              dynamic_cast<BroadcastOp*>(squeeze->in()->definition())) {
+        if (bcast->getBroadcastDimFlags() == squeeze->getSqueezeDimFlags() &&
+            bcast->out()->uses().size() == 1) {
```

Comment:
I don't think this check is needed. Even when `bcast->out()` has multiple uses, we can still replace `squeeze->out()` without affecting other paths. If you agree on that, you may also want to add one more unit test for that case.

Reply:
agree. added `BcastSqueezeMultipleUses`